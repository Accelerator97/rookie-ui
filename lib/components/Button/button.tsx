import classNames from 'classnames'
import React,{ButtonHTMLAttributes,AnchorHTMLAttributes,FC} from 'react'

export type ButtonSize = 'lg' | 'sm'
export type ButtonType = 'primary' | 'default' | 'danger' | 'link'

interface BaseButtonProps {
    className?: string;
    disabled?: boolean;
    size?: ButtonSize;
    btnType?: ButtonType;
    children: React.ReactNode;
    /**链接按钮跳转地址 */
    href?: string
}
type NatvieButtonProps = ButtonHTMLAttributes<HTMLElement> & BaseButtonProps
type AnchorButtonProps = AnchorHTMLAttributes<HTMLElement> & BaseButtonProps
export type ButtonProps = Partial<NatvieButtonProps & AnchorButtonProps>

export const Button: FC<ButtonProps> = (props) => {
    const { btnType, disabled, size, children, href ,className,...restprops} = props
    const classes = classNames('btn',className, {
        [`btn-${btnType}`]: btnType,
        [`btn-${size}`]: size,
        'disabled': (btnType === 'link') && disabled
    })
    if (btnType === 'link' && href) {
        return (
            <a className={classes} href={href} {...restprops}>{children}</a>
        )
    } else {
        return (
            <button className={classes} disabled={disabled} {...restprops}>{children}</button>
        )
    }
}
Button.defaultProps={
    disabled:false,
    btnType:'default'
}

export default Button;
