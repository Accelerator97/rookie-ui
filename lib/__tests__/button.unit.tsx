import * as rendrer from 'react-test-renderer'
import * as React from 'react'
import Button from '..//button'
describe('button',()=>{
    it('是个Div',()=>{
        const tree = rendrer.create(<Button></Button>).toJSON()
        expect(tree).toMatchSnapshot()
    })
})