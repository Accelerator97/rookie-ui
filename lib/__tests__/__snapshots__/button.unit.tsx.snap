// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button 是个Div 1`] = `
<div>
  按钮
</div>
`;
